ql/termstructures/yield/ratehelpers.hpp

```diff
@@ -347,6 +358,7 @@ namespace QuantLib {
         Handle<YieldTermStructure> discountHandle_;
         RelinkableHandle<YieldTermStructure> discountRelinkableHandle_;
         ext::optional<bool> useIndexedCoupons_;
+        bool swapBuiltViaGivenBuilder;
```

Comment:
We are supporting two different ctor types
1. one accepting arguments (e.g. `Period tenor`) to built up the `MakeVanillaSwap`
2. another one accepting directly the builder, i.e. `MakeVanillaSwap swapBuilder`

Thus, different member variables are used. We could represent this in code (e.g. via `optional`, `variant`). Here I used a version of this single boolean for minimal code changes.